src/main/ruby/core/rbconfig.rb

```diff
@@ -160,8 +160,14 @@ module RbConfig
   end
 
   def self.ruby
-    Truffle::Boot.ruby_launcher or
-        raise "we can't find the TruffleRuby launcher - set -Xlauncher=, or your launcher should be doing this for you"
+    @ruby_launcher ||= begin
+      # ruby launcher is properly set
+      Truffle::Boot.ruby_launcher ||
+          # determine launcher from ruby_home
+          ("#{CONFIG['bindir']}/#{CONFIG['ruby_install_name']}" if Truffle::Boot.ruby_home) ||
```

Comment:
Would this be clearer as:
```ruby
(CONFIG['bindir'] && "#{CONFIG['bindir']}/#{CONFIG['ruby_install_name']}") ||
```
?

Reply:
I don't think so, It hides what it really depends on, which is `Truffle::Boot.ruby_home` being present.

Reply:
Right. I also meant the `&&` vs `if` which seems a bit odd in a conditional chain like this.